### PR TITLE
Adds an endpoint for retrieving a SanitizedSecret by name

### PR DESCRIPTION
**Pull request by stfinney**
This should probably eventually replace secretInfo, which returns a SecretDetailResponseV2. This is the only place SecretDetailResponseV2 is used, and it would be nice to get rid of it.

### CODE REVIEW

File: server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java [hunk 1]
```diff
@@ -218,6 +218,36 @@ public void partialUpdateSecret_notFound() throws Exception {
   }
 
   //---------------------------------------------------------------------------------------
+  // getSanitizedSecret
+  //---------------------------------------------------------------------------------------
+
+  @Test public void getSanitizedSecret_notFound() throws Exception {
+    Request get = clientRequest("/automation/v2/secrets/non-existent").get().build();
```
Comment (jbpeirce) [inline server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java]: This should test `"/automation/v2/secrets/non-existent/sanitized"`

File: server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java [hunk 2]
```diff
@@ -945,6 +975,13 @@ SecretDetailResponseV2 lookup(String name) throws IOException {
     return mapper.readValue(httpResponse.body().byteStream(), SecretDetailResponseV2.class);
   }
 
+  SanitizedSecret lookupSantizedSecret(String name) throws IOException {
```
Comment (jbpeirce) [inline server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java]: nit: `lookupSanitizedSecret`

File: server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java (hunk 1 above)
Comment (stfinney) [inline server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java]: Fixed

File: server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java (hunk 2 above)
Comment (stfinney) [inline server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java]: Fixed

**Comment by jbpeirce** [comment]
Looks good!

**Pull request merged**